source/uwp/Renderer/lib/AdaptiveElementParserRegistration.h

```diff
@@ -60,4 +62,39 @@ namespace AdaptiveCards { namespace Uwp
     private:
         Microsoft::WRL::ComPtr<AdaptiveCards::Uwp::AdaptiveElementParserRegistration> m_parserRegistration;
     };
+
+    template<
+        typename TAdaptiveCardElement,
+        typename TSharedModelElement,
+        typename TSharedModelParser>
+        HRESULT FromJson(
+            ABI::Windows::Data::Json::IJsonObject* jsonObject,
+            ABI::AdaptiveCards::Uwp::IAdaptiveElementParserRegistration* elementParserRegistration,
+            ABI::AdaptiveCards::Uwp::IAdaptiveActionParserRegistration* actionParserRegistration,
+            ABI::AdaptiveCards::Uwp::IAdaptiveCardElement** element)
+    {
+        std::string jsonString;
+        JsonObjectToString(jsonObject, jsonString);
+
+        std::shared_ptr<AdaptiveCards::ElementParserRegistration> sharedModelElementParserRegistration;
+        ComPtr<AdaptiveElementParserRegistration> elementParserRegistrationImpl = PeekInnards<AdaptiveElementParserRegistration>(elementParserRegistration);
+        if (elementParserRegistrationImpl != nullptr)
```

Comment:
>nullptr [](start = 45, length = 7)

What happens if either registration ends up as null? I don't believe the shared model parsers' DeserializeFromString does null checks on the registrations.

Reply:
Updated this code to create one if there isn't one, although i toyed with the idea of just returning invalid arg in this case, since we expect this to be called from a custom parser where we will have just passed them the registrations.

---
In reply to: [147445256](https://github.com/Microsoft/AdaptiveCards/pull/854#discussion_r147445256) [](ancestors = 147445256)